### ISSUE TITLE
Implement a fuzzy dir finder

### ISSUE THREAD

**Issue by eburghar** [helix-editor/helix#4430]
I use extensively a fuzzy dir finder in kakoune I made with skim to quickly cd into a project directory and then fuzzy pick a file from there.

I have the option to start the finder from a given dir (by default it fuzzy finds from the current directory).

So there could be a fuzzy dir picker accessible from the space mode, and a command accepting argument with completion mode on the prompt mode, but we can rely on the `:cd` command for that and then pick in space mode in 2 steps.

It should ne reasonably easy to implement on top of the actual file picker (choose dir instead of file, cd instead of open).

What do you think ?

**Comment by the-mikedavis** [comment]
Does https://github.com/helix-editor/helix/pull/2412 cover what you're looking for?

**Comment by eburghar** [comment]
> Does #2412 cover what you're looking for?

I think this is about a different kind file picker. I was talking about a dir picker which does a cd at the end. The `:cd` command completion is not fuzzy and limited to one level which force you to press tab many times at each level. With a cd fuzzy picker you can select the dir with just a few keystrokes independently of its depth in the current hierarchy.

**Comment by eburghar** [comment]
> Hey @eburghar I think something like this can also help. What you think?
> 
> ![image](https://user-images.githubusercontent.com/11697220/197837743-b78712af-346e-46b9-85c2-2cd356b014e9.png)

Yes that would be better. I just fear that it would offer less readability than a vertical list like the file picker. You should also have the fuzzy pattern highlighted in candidates.
